### PR TITLE
20016 - update missed alternate name instances

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#20016

*Description of changes:*

Updated missed instances with generic business object

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by PaulGarewal — APPROVED** [review]
Looks good from what I can tell

File: legal-api/src/legal_api/reports/report.py [hunk 1]
```diff
@@ -440,12 +440,12 @@ def _format_address(address):
         return address
 
     @staticmethod
-    def _populate_business_info_to_filing(filing: Filing, legal_entity: LegalEntity):
-        founding_datetime = LegislationDatetime.as_legislation_timezone(legal_entity.founding_date)
+    def _populate_business_info_to_filing(filing: Filing, business: any):
+        founding_datetime = LegislationDatetime.as_legislation_timezone(business.founding_date)
```
Comment (leodube-aot) [inline legal-api/src/legal_api/reports/report.py:444]: I don't think AlternateName entities have a `founding_date` attribute. Maybe we do [something like this](https://github.com/bcgov/lear/pull/2522/files#diff-21be2c43c7121075d95bf3a989eff70fec5a03fa599fe0f5217d2c565cd8d6a1R359) and use `start_date` for alternate name entities?

**Review by leodube-aot — APPROVED** [review]

**Review by JazzarKarim — APPROVED** [review]

File: legal-api/src/legal_api/resources/v2/internal_services.py [hunk 2]
```diff
@@ -36,39 +36,40 @@ def update_bn_move():
...
 
-    legal_entity = LegalEntity.find_by_tax_id(old_bn)
-    if legal_entity:
-        legal_entity.tax_id = new_bn
-        legal_entity.save()
+    for identifier in json_input.keys():
+        business = business_service.fetch_business(identifier)
+        if business and business.is_legal_entity:
+            business.tax_id = json_input[identifier]
+            business.save()
```
Comment (argush3) [inline legal-api/src/legal_api/resources/v2/internal_services.py:43]: i'm not sure about this code.  

Why do we need to loop through the keys?  Have the implementation change in `main`?

Comment (ketaki-deodhar) [reply legal-api/src/legal_api/resources/v2/internal_services.py:43]: Nothing has changed

File: legal-api/src/legal_api/services/business_service.py [hunk 3]
```diff
@@ -67,3 +67,26 @@ def fetch_business_by_filing(filing):  # pylint: disable=redefined-builtin
             return alternate_name_entity
 
         return None
+
```
Comment (ketaki-deodhar) [inline legal-api/src/legal_api/services/business_service.py:70]: Fetches business by tax_id or bn15

File: legal-api/src/legal_api/resources/v2/internal_services.py [hunk 4]
```diff
@@ -36,39 +36,50 @@ def update_bn_move():
     if not json_input or not (old_bn := json_input.get("oldBn")) or not (new_bn := json_input.get("newBn")):
         return ({"message": "No oldBn or newBn in body of post."}, HTTPStatus.BAD_REQUEST)
 
-    legal_entity = LegalEntity.find_by_tax_id(old_bn)
-    if legal_entity:
-        legal_entity.tax_id = new_bn
-        legal_entity.save()
+    business = business_service.fetch_business_by_tax_id(old_bn)
```
Comment (ketaki-deodhar) [inline legal-api/src/legal_api/resources/v2/internal_services.py:39]: Fetch business by tax_id or bn15 and set newBn

File: legal-api/tests/unit/resources/v2/test_internal_services.py [hunk 5]
```diff
@@ -46,6 +47,39 @@ def test_update_bn_move(session, client, jwt):
             assert LegalEntity.find_by_tax_id(new_bn)
 
 
+def test_update_bn_move_sp(session, client, jwt):
```
Comment (argush3) [inline legal-api/tests/unit/resources/v2/test_internal_services.py:50]: thanks for the SP test!

**Review by argush3 — APPROVED** [review]

**Pull request merged**